test/client/submit.js

```diff
@@ -206,93 +207,122 @@ module.exports = function() {
       });
     });
 
-    it('can create a new doc then fetch', function(done) {
-      var doc = this.backend.connect().get('dogs', 'fido');
-      doc.create({age: 3}, function(err) {
-        if (err) return done(err);
-        doc.fetch(function(err) {
-          if (err) return done(err);
-          expect(doc.data).eql({age: 3});
-          expect(doc.version).eql(1);
-          done();
-        });
+
+    describe('create', function() {
+      describe('metadata enabled', function() {
+        runCreateTests();
       });
-    });
 
-    it('calling create on the same doc twice fails', function(done) {
-      var doc = this.backend.connect().get('dogs', 'fido');
-      doc.create({age: 3}, function(err) {
-        if (err) return done(err);
-        doc.create({age: 4}, function(err) {
-          expect(err).instanceOf(Error);
-          expect(doc.version).equal(1);
-          expect(doc.data).eql({age: 3});
-          done();
+      describe('no snapshot metadata available', function() {
+        beforeEach(function() {
+          var getSnapshot = MemoryDB.prototype.getSnapshot;
+          sinon.stub(MemoryDB.prototype, 'getSnapshot')
+            .callsFake(function(collection, id, fields, snapshotOptions, callback) {
+              // The $submit field requests metadata, so setting to false will
+              // imitate a driver that doesn't support metadata; or a snapshot that's
+              // missing it
+              if (fields) fields.$submit = false;
+              getSnapshot.call(this, collection, id, fields, snapshotOptions, callback);
```

Comment:
I'm a bit worried `$submit` has or in the future will have other side effects, so it'd be safer to manually strip the metadata from the resulting snapshot